src/Sylius/Bundle/CartBundle/Controller/CartItemController.php

```diff
@@ -137,6 +140,30 @@ public function removeAction(Request $request)
     }
 
     /**
+     * @param Request $request
+     * @param CartInterface $cart
+     */
+    protected function saveCartInSession(Request $request, CartInterface $cart)
+    {
+        $session = $request->getSession();
+
+        if (!$session->has($this->getSessionCartPrefix())) {
```

Comment:
Are you sure? `$session->has('_sylius.cart')` should always return `false` as you do `$session->set('_sylius.cart.CHANNEL_CODE', ...)` instead.